ivre/db/mongo.py

```
@@ -390,6 +400,43 @@ def getscan(self, scanid, archive=False, **kargs):
         return self.db[self.colname_scans].find_one(
             {'_id': scanid}, **kargs)
 
+    def getscreenshot(self, port):
+        """Returns the content of a port's screenshot."""
+        url = port.get('screenshot')
+        if url is None:
+            return None
+        if url == "field":
+            return port.get('screendata')
+
+    def setscreenshot(self, host, port, data, archives=False,
+                      overwrite=False):
+        """Sets the content of a port's screenshot."""
+        port = [p for p in host['ports'] if p['port'] == port][0]
```

You should raise a proper exception if the port doesn't exist